server/src/main/java/server/bazel/bazelWorkspaceAPI/WorkspaceAPI.java

```diff
@@ -0,0 +1,213 @@
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param targetPath The path, represented by a String, to the given build target
+     *          expected format "//path/to/file.java"
+     * @return true if the sourcefile is stored in the workspace tree at targetPath location
+     */
+    public boolean isSourceFileInPackage(String targetPath){
+        Package packageFromPath;
+        try {
+            packageFromPath =  findNodeOfGivenPackagePath(PathType.FILE_PATH,targetPath).getValue();
+        } catch (WorkspaceAPIException e) {
+            return false;
+        }
+        List<SourceFile> sourceFiles = packageFromPath.getSourceFiles();
+        for(SourceFile target: sourceFiles){
+            String sourcePath = target.getPath().toString();
+            if(sourcePath.equals(targetPath)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Private methods to be used by the the API
+
+    /**
+     *
+     * @param path Accepts a path in the given format "//path/to/package"
+     * @return the WorkspaceTree node that represents the given package
+     * @throws WorkspaceAPIException if path is not in given workspace tree
+     */
+    private Node findNodeOfGivenPackagePath(PathType type,String path) throws WorkspaceAPIException {
+        Node lastNode;
+
+        lastNode = workspaceTree.getRoot();
+        String[] packages = getPackageAsAnArray(type, path);
+
+        for(int i = 1; i < packages.length; i ++){
+            Optional<WorkspaceTree.Node> potentialNode = lastNode.getChild(packages[i]);
+            if(potentialNode.isEmpty()){
+                throw new WorkspaceAPIException("Invalid Path");
+            } else {
+                lastNode = lastNode.getChild(packages[i]).get();
+            }
+        }
+
+        return lastNode;
+    }
+
+    /**
+     *
+     * @param type an enumerator to represent TargetPath, OpenPath, FilePath
+     * @param givenPath The path to the given target, package, or file
+     *          expected format "//path/to:target" or "//path/to/package" or "//path/to/file.java"
+     * @return An Array of strings representing the package
+     *          expected format ["path", "to", "package"]
+     *          TargetPaths and filePath returns will not contain the file.java or :target int he return array
+     * @throws WorkspaceAPIException If filepath is bad, does not start at root, or doesn't match given type
+     */
+    private String[] getPackageAsAnArray(PathType type, String givenPath) throws WorkspaceAPIException {
+        // Assert that the root path was passed, may need to be variable based on operating system.
+        if (givenPath == null){
+            throw new WorkspaceAPIException("Given File Path is null");
+        }
+        if (givenPath.length() < 2){
+            throw new WorkspaceAPIException("Given File Path does not start at Root");
+        }
+        if(givenPath.charAt(0) != '/' |
```

Comment:
I think this logic should be moved to it's own domain object in `server.bazel.tree.Target` (or something)

```java
/*
 * Represents a //path/to/bazel:target string
 * or represents //path/to/some/source/file.java
 * or represents //path/to/something
 */
@immutable
public class Target {
  public Target(String path) {
    this.path = path
  }

  public String getPackageName() { ... }

  public String getWorkspacePath() { ... }

  // etc...
}
```

Reply:
Again, `BuildTarget` exists

Reply:
Then should we use it here instead of strings?

Reply:
Abstracted out to BuildTarget object